### ISSUE TITLE
with PYRO, solver_manager.queue(model, opt=optsolver)  ignores executable path of optsolver

### ISSUE THREAD

**Issue by maajdl** [Pyomo/pyomo#565]
Hello,

I wanted to test different versions of Ipopt on the same model.
I also wanted to do that in parallel  using PYRO .
The script looks like this:

```
solver_manager = SolverManagerFactory('pyro')

for exepath in ipoptExecutables:
    instance = mymodel()
    optsolver = SolverFactory('ipopt', executable=exepath )
    action = solver_manager.queue(instance, opt=optsolver)
```

When observing the results, it appears that   solver_manager   never uses the path indicated in optsolver.
Instead it uses the standard path where Ipopt usually resides.

Is that by design, or is that a bug?
Is there some workaround or some alternative way?

Thanks,

Michel

**Comment by qtothec** [comment]
You can try using the multiprocessing package as an option.

**Comment by maajdl** [comment]
Thanks qtothec

I guess this is this package:   https://docs.python.org/2/library/multiprocessing.html   ?

I had already tried the "queues and threading"  objects from Python but was stopped by an hopeless error.

I will try multiprocessing, it looks simple at first sight. 
Lets's hope it works.
I need to see how I could catch the first result and stop then the other processes that would come later. 
My purpose is to avoid waiting for a solver that will fail or reach maximum iterations, or take a long time.

Thanks,

Michel

**Comment by qtothec** [comment]
@maajdl - you can start a bunch of parallel jobs and poll for a solution. Once one is found, you can kill the rest of the worker pool. That's one idiom that would work. This sounds like a good start on a parallel multi-start solver, if that's something that you'd be interested in contributing back to the Pyomo codebase.

**Comment by maajdl** [comment]
Thanks qtohec,

I finally used the   threading  and  queue   packages  for a 10-liners solution.
I  had to replace the signature in   pyutilib\subprocess\processmngr.py   as follows to get it working with Pyomo:

```
run_command( ..., define_signal_handlers=None  , ...)             
run_command( ..., define_signal_handlers=False , ...)
```

Thanks again,

Michel

**Comment by blnicho** [comment]
@maajdl looks like your question was answered so I'm closing this.

**Comment by maajdl** [comment]
Maybe I would still be interrested to know a better place to modify the define_signal_handlers .
Doing it in  pyutilib  in the signature of  processmngr.run_command  could affect other programs.
Maybe I should modify at the place where Pyomo call this method.
Thanks,
Michel

**Comment by qtothec** [comment]
@maajdl, that might be a new ticket. I must say that I'm not familiar with the part of the codebase that includes `define_signal_handlers`.
